core/node/api_server/src/execution_sandbox/vm_metrics.rs

```diff
@@ -156,24 +152,25 @@ pub(super) fn collect_tx_execution_metrics(
         .sum();
 
     TransactionExecutionMetrics {
-        initial_storage_writes: writes_metrics.initial_storage_writes,
-        repeated_storage_writes: writes_metrics.repeated_storage_writes,
-        gas_used: result.statistics.gas_used as usize,
+        writes,
+        vm: VmExecutionMetrics {
+            gas_used: result.statistics.gas_used as usize,
+            published_bytecode_bytes,
+            l2_l1_long_messages,
+            l2_to_l1_logs: result.logs.total_l2_to_l1_logs_count(),
+            user_l2_to_l1_logs: result.logs.user_l2_to_l1_logs.len(),
+            contracts_used: result.statistics.contracts_used,
+            // FIXME: incorrectly defined? (a number of factory deps, not number of contracts deployed)
+            contracts_deployed,
```

Comment:
Subj. I'm not 100% sure this is a mistake, but if we indeed want to have the number of factory deps as a stat, (a) the name is misleading; (b) the stats can be extracted from transaction data, it's not really *VM execution* metrics.

Reply:
IDK, I'd check if this data is used anywhere at all. If not, probably it's just in case we want to look at stats -- and in this case we can remove it.

Reply:
It's not used in the Era codebase, and AFAIU isn't used in the test node / zksync-foundry either. So is it safe to remove? This would allow to simplify metrics construction.

Reply:
It is